### ISSUE TITLE
What is the use of the '*__asm' pseudo-attribute on JS library functions?

### ISSUE THREAD

**Issue by Akaricchi** [emscripten-core/emscripten#9720]
I thought this means the function is to be treated as asm.js code (and perhaps compiled into wasm), but it seems to have no effect on my custom library functions whatsoever. Should I use it and when? Is there any documentation about implementing JS libraries in general?

**Comment by kripken** [comment]
This attribute used to mean the function is hand-written asm.js. But it was not very useful as its hard to write asm.js by hand, so we used it rarely.

In any case, with the new wasm backend, there is no asm.js support, so the flag has no effect. We'll remove it when we completely switch over.

We should probably have more docs on writing JS libraries - sorry about that! The best we have is really the core system JS libraries as example code, I guess. Would be good to improve.

**Comment by Akaricchi** [comment]
Thanks for the clarification. I guess this implies that attempting to maintain all the asm.js annotations like `x|0` and using macros like `makeSetValueAsm` is pointless, correct?

>We should probably have more docs on writing JS libraries - sorry about that! The best we have is really the core system JS libraries as example code, I guess. Would be good to improve.

Yeah, that's what I figured. I came across the `_asm` attribute and asm.js annotations quite a few times in library code, which made me wonder about this.

What about `_sig`, by the way? Is it also ignored or is it used for some sort of validation/debugging purposes?
